### PR TITLE
refactor(cbdc-example): added gateways and contracts

### PR DESCRIPTION
**Pull request by LordKubaya**
- backend refactored

### CODE REVIEW

File: examples/cactus-example-cbdc-bridging-backend/cache/solidity-files-cache.json [hunk 1]
```diff
@@ -0,0 +1 @@
...ler":true},"src/solidity/main/satp-wrapper.sol":{"lastModificationDate":1725357617051,"contentHash":"3796cb4be7c06c5c8912e8137293b8a9","sourceName":"src/solidity/main/satp-wrapper.sol","compilerSettings":{"solc":{"optimizer":{"enabled":true,"runs":200},"metadata":{"useLiteralContent":false,"bytecodeHash":"ipfs","appendCBOR":true},"outputSelection":{"*":{"*":["abi","evm.bytecode","evm.deployedBytecode","evm.methodIdentifiers","metadata"]}},"evmVersion":"paris","viaIR":false,"libraries":{}},"vyper":{"evmVersion":"paris","outputSelection":{"*":{"*":["abi","evm.bytecode","evm.deployedBytecode"]}}}},"imports":["node_modules/@openzeppelin/contracts/access/Ownable.sol","node_modules/@openzeppelin/contracts/utils/Context.sol","node_modules/@openzeppelin/contracts/utils/Strings.sol","node_modules/@openzeppelin/contracts/utils/math/Math.sol","node_modules/@openzeppelin/contracts/utils/math/SignedMath.sol","src/solidity/main/ITraceableContract.sol"],"versionRequirement":"^0.8.20","artifacts":{"SATPWrapperContract":{"0.8.20":{"path":"satp-wrapper.sol/SATPWrapperContract.json","build_id":"533d2c5e3c1a73b85c754b76cd8b9ca2"}}},"seenByCompiler":true}},"builds":["533d2c5e3c1a73b85c754b76cd8b9ca2"]}
```
Comment (petermetz) [inline examples/cactus-example-cbdc-bridging-backend/cache/solidity-files-cache.json:1]: @LordKubaya I recommend ignoring build cache artifacts like this in the gitignore file

File: examples/cactus-example-cbdc-bridging-backend/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Change Log
```
Comment (petermetz) [inline examples/cactus-example-cbdc-bridging-backend/CHANGELOG.md:1]: @LordKubaya I recommend deleting this file - changelog.md files are generated by the release automation at the time when we issue releases (I'm guessing this just stuck around as a copy paste thing)

**Review by petermetz — COMMENTED** [review]
LGTM with comments

**Review by RafaelAPB — COMMENTED** [review]
LGTM. Please address Peter's suggestions. Also take a look at GitGuardian. Thanks

File: examples/cactus-example-cbdc-bridging-backend/cache/solidity-files-cache.json (hunk 1 above)
Comment (LordKubaya) [reply examples/cactus-example-cbdc-bridging-backend/cache/solidity-files-cache.json]: Done, thank you!

File: examples/cactus-example-cbdc-bridging-backend/CHANGELOG.md (hunk 2 above)
Comment (LordKubaya) [reply examples/cactus-example-cbdc-bridging-backend/CHANGELOG.md]: yeah that was it

**Comment by RafaelAPB** [comment]
Merged as feedback was incorporated

**Pull request merged**